lib/main_common.pm

```diff
@@ -1600,50 +1600,74 @@ sub load_filesystem_tests {
-        loadtest 'wicked/advanced/' . $f . '/t07_tun_interface_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t08_tun_interface_xml';
-        loadtest 'wicked/advanced/' . $f . '/t09_tap_interface_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t10_tap_interface_xml';
-        loadtest 'wicked/advanced/' . $f . '/t11_bridge_interface_legacy';
-        loadtest 'wicked/advanced/' . $f . '/t12_bridge_interface_xml';
+        return qw(
+          t01_gre_tunnel_legacy
+          t02_gre_tunnel_xml
+          t03_sit_tunnel_legacy
+          t04_sit_tunnel_xml
+          t05_ipip_tunnel_legacy
+          t06_ipip_tunnel_xml
+          t07_tun_interface_legacy
+          t08_tun_interface_xml
+          t09_tap_interface_legacy
+          t10_tap_interface_xml
+          t11_bridge_interface_legacy
+          t12_bridge_interface_xml
+        );
     }
     elsif (check_var('WICKED', 'startandstop')) {
-        my $f = get_var('IS_WICKED_REF') ? 'ref' : 'sut';
-        loadtest 'wicked/startandstop/' . $f . '/t01_standalone_card_ifdown_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t02_bridge_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t03_bridge_ifup_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t04_bridge_ifup_remove_all_config_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t05_bridge_ifup_remove_one_config_ifreload';
-        loadtest 'wicked/startandstop/' . $f . '/t06_bridge_ifdown_create_new_config_ifreload_ifdown_ifup';
-        loadtest 'wicked/startandstop/' . $f . '/t07_bridge_ifdown_remove_one_config_ifreload_ifdown_ifup';
-        loadtest 'wicked/startandstop/' . $f . '/t08_sit_tunnel_ifdown';
-        loadtest 'wicked/startandstop/' . $f . '/t09_openvpn_tunnel_ifdown';
-        loadtest 'wicked/startandstop/' . $f . '/t10_vlan_ifup_all_ifdown_one_card';
+        return qw(
+          t01_standalone_card_ifdown_ifreload
+          t02_bridge_ifreload
+          t03_bridge_ifup_ifreload
+          t04_bridge_ifup_remove_all_config_ifreload
+          t05_bridge_ifup_remove_one_config_ifreload
+          t06_bridge_ifdown_create_new_config_ifreload_ifdown_ifup
+          t07_bridge_ifdown_remove_one_config_ifreload_ifdown_ifup
+          t08_sit_tunnel_ifdown
+          t09_openvpn_tunnel_ifdown
+          t10_vlan_ifup_all_ifdown_one_card
+        );
     }
     else {
         die 'Unhandled WICKED test selection: ' . get_var('WICKED');
     }
 }
 
+sub load_wicked_barrier {
+    my $args         = OpenQA::Test::RunArgs->new();
+    my @wicked_tests = qw(before_test);
+    push(@wicked_tests, get_wicked_tests());
+    $args->{wicked_tests} = \@wicked_tests;
+    loadtest('wicked/barrier_init', run_args => $args);
+}
+
+sub load_wicked_tests {
+    my $dir = get_var('IS_WICKED_REF') ? 'ref' : 'sut';
+    my $suite = get_var('WICKED');
+
+    die "Unsupported WICKED suite: $suite" unless (grep { $_ eq $suite } qw(basic advanced startandstop));
```

Comment:
I wonder if these could be kept in a file or in a  hashref, so that if a new suite needs to be added you don't have to edit in 2 different places. Something like a config file or it's own module... (maybe I'm just thinking out loud):

`` 
my $tests = {
   basic => [a b c],
   advanced => [d e f]
};
...
die "Unsupported WICKED suite: $suite" unless (grep { $_ eq $suite } keys %{$tests}
```

Reply:
Agree, we thought about getting the list of tests directly from the fs.
Using `$bmwqemu::vars{CASEDIR};` as base dir and depending on `'/tests/' . get_var('WICKED') . '/' . $dir` the list of tests.

Reply:
That's a very valid suggestion, as we grow with number of suites, we will have to change this again and again. Since this PR is getting very complex, I propose to create a ticket with your thoughts and work on it for another PR. Is that ok for you?

Reply:
let's try to not save the world with single PR :) so yes ideas are valid but we can do it later in another PR